### PR TITLE
Slash is html encoded in csv file #4622

### LINKED ISSUE #4622: Slash is html encoded in csv file
Received this bug report from a user:
My student name has D/O in her name but displayed as D&amp;#x2f;O  in the csv file.

### PR DESCRIPTION
**Pull request by chen-zhuo**
Fixes #4622.

When sanitizing people's names for csv files, remember to call `StringHelper.recoverFromSanitizedText()` method.

My Concerns:
1. I only changed the codes that caused issue #4622. However, I saw codes that look like `Sanitizer.sanitizeForCsv(humanName)` somewhere else as well. Should I change them accordingly to `Sanitizer.sanitizeForCsv(StringHelper.recoverFromSanitizedText(humanName))`?
2. Alternatively, should I just call `StringHelper.recoverFromSanitizedText()` inside `Sanitizer.sanitizeForCsv()`?

### CODE REVIEW

**Comment by wkurniawan07** [comment]
I don't think this is how it's supposed to be solved. The problem lies in `FeedbackSessionResultsBundle`, the methods `getNameForEmail`, `getLastNameForEmail`, and perhaps `getTeamNameForEmail`, all of which wrongly used `sanitizeForHtml`.
If there are usages of the said methods that need HTML sanitization, `sanitizeForHtml` should be called separately.

**Comment by chen-zhuo** [comment]
@wkurniawan07, I have edited the codes as you instructed. This does solve issue #4622. However, this might have side effects. Existing codes calling `getNameForEmail()`, `getLastNameForEmail()` or `getTeamNameForEmail()` may expect the string to be html-sanitized already. Could you advise what should I do next?

**Comment by wkurniawan07** [comment]
I'm afraid this is your homework :p See which usages need HTML sanitization and which ones don't. In general, you only need it if you're processing for HTML display.

**Comment by chen-zhuo** [comment]
@wkurniawan07, I tried to study the call hierarchy of `getNameForEmail()` method. There are many members calling it, but it seems that they all point to the `sectionPanels` object class. I have made some changes accordingly. Please advise the correctness. Did I miss out anything?

**Comment by wkurniawan07** [comment]
Any place where the other two methods are called for HTML display?

**Comment by chen-zhuo** [comment]
@wkurniawan07, I tried to find all members calling that three methods for HTML display. Did I miss out anything?

**Comment by wkurniawan07** [comment]
This issue is more wide-reaching than I thought. I'll do some check and will get back later.

**Comment by thyageshm** [comment]
@wkurniawan07 any updates on this issue?

**Comment by wkurniawan07** [comment]
I'd say you have done enough here, @bamboo3250 will fill up the remaining gaps in #5058.
@bamboo3250, you can have a quick look here.

File: src/main/java/teammates/common/datatransfer/FeedbackNumericalScaleQuestionDetails.java [hunk 1]
```diff
@@ -383,13 +383,13 @@ private String getDisplayableRecipientName(boolean isHiddenRecipient,
         
         // Replace current user name with "You"
         if (!isHiddenRecipient && isRecipientCurrentUser && hasAtLeastTwoResponses){
-            return isRecipientTypeStudent? "You" : "Your Team (" + currentUserTeam + ")";
+            return isRecipientTypeStudent? "You" : "Your Team (" + Sanitizer.sanitizeForHtml(currentUserTeam) + ")";
```
Comment (bamboo3250) [inline src/main/java/teammates/common/datatransfer/FeedbackNumericalScaleQuestionDetails.java]: We shouldn't sanitize inside this function. Ideally it should be right before injecting into HTML.

File: src/main/java/teammates/common/datatransfer/FeedbackNumericalScaleQuestionDetails.java [hunk 2]
```diff
@@ -401,13 +401,13 @@ private String getDisplayableRecipientTeam(boolean isHiddenRecipient,
         
         // Replace current user team with "" when recipient type is not student
         if (!isHiddenRecipient && isRecipientCurrentUser && hasAtLeastTwoResponses){
-            return isRecipientTypeStudent? currentUserTeam : "";
+            return isRecipientTypeStudent? Sanitizer.sanitizeForHtml(currentUserTeam) : "";
```
Comment (bamboo3250) [inline src/main/java/teammates/common/datatransfer/FeedbackNumericalScaleQuestionDetails.java]: Similarly for these two.

**Comment by bamboo3250** [comment]
Just added some comments. Those are not critical and will be resolved in my PR anyway so I think you still can merge it.

**Comment by wkurniawan07** [comment]
Noted. I'll notify again when this gets merged so you can sync your PR.

**Comment by bamboo3250** [comment]
@wkurniawan07 thanks :+1:

**Pull request merged**